perceval/backends/public_inbox/public_inbox.py

```diff
@@ -0,0 +1,301 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2023 Bitergia
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Authors:
+#     Jose Javier Merchante <jjmerchante@bitergia.com>
+#
+
+import email
+import logging
+import re
+
+from grimoirelab_toolkit.datetime import str_to_datetime, InvalidDateError
+
+from ...utils import (DEFAULT_DATETIME,
+                      DEFAULT_LAST_DATETIME,
+                      message_to_dict)
+from ...errors import ParseError
+from ...backend import (BackendCommand,
+                        BackendCommandArgumentParser)
+from ...backends.core.git import (Git,
+                                  GitRepository)
+
+
+CATEGORY_MESSAGE = 'message'
+
+logger = logging.getLogger(__name__)
+
+
+class PublicInbox(Git):
+    """PublicInbox backend for Perceval.
+
+    This class retrieves the email messages stored in one public-inbox
+    mirror repository. Initialize this class passing the directory path
+    of the repository where the file is stored. The origin of the data
+    will be set to the value of `uri`.
+
+    :param uri: URI of the public-inbox; typically, the URL of their
+    mailing list
+    :param gitpath: directory path of the public-inbox repository
+    :param tag: label used to mark the data
+    :param archive: archive to store/retrieve items
+    """
+    version = '0.1.0'
+
+    CATEGORIES = [CATEGORY_MESSAGE]
+
+    DATE_FIELD = 'Date'
+    MESSAGE_ID_FIELD = 'Message-ID'
+
+    def __init__(self, uri, gitpath, tag=None, archive=None):
+        # Initialize the backend using the Git backend
+        super().__init__(uri, gitpath, tag=tag, archive=archive)
+
+    def fetch(self, category=CATEGORY_MESSAGE, from_date=DEFAULT_DATETIME, to_date=DEFAULT_LAST_DATETIME):
```

Comment:
`from_date` filters by commit date, if I'm not wrong. Is the commit date the same date when the message was sent?

Reply:
Yes, that's a problem users can have. That's why we implemented the `--latest-items` option. It updates the repository and retrieves only the new commits. Maybe we can implement something similar. That will mean to retrieve the commits from the latest repo on the alternatives list...but I don't know if that's something possible we can do.